### ISSUE TITLE
Migrate class labels from experiments.config to src without behavior changes

### ISSUE THREAD

**Issue by t22cs011** [t22cs011/braindecodetest#280]
## Goal

Remove the newly explicit `src.models.registry -> experiments.config.classes` dependency by introducing a small `src/` source of truth for BCI2020 class labels, without changing any runtime behavior.

This is the next narrow follow-up after PR #279, which moved `MODEL_SPECS` from `experiments.models` to `src.models.registry` while preserving the remaining class-count dependency on `experiments.config.classes` as an explicit temporary boundary.

## Background

After PR #279, the model registry now lives in:

```text
src/models/registry.py
```

The legacy compatibility module remains:

```text
experiments/models.py
```

and re-exports:

```python
from src.models.registry import MODEL_SPECS
```

However, `src.models.registry` still imports:

```python
from experiments.config import classes
```

This is behavior-preserving and intentionally visible in `tests/test_src_experiments_dependency_boundary.py`, but it leaves `src.models.registry` dependent on `experiments.config` only to calculate `n_outputs = len(classes)`.

That dependency is small and safer to migrate before touching broader config/data/cache/training-loop surfaces.

## Scope

- Add a small `src/` module for BCI2020 class labels, for example:

```text
src/config/labels.py
```

or another clearly named `src/` location if more appropriate.

- Define the current class labels there with the exact same order and values currently exposed as `experiments.config.classes`.

- Update `src/models/registry.py` to import the labels from the new `src/` module, not from `experiments.config`.

- Keep `experiments.config.classes` available and behavior-compatible. If feasible, make it re-use the new `src/` constant rather than duplicating the labels.

- Update `tests/test_src_experiments_dependency_boundary.py` to remove the temporary allowance:

```text
src/models/registry.py -> experiments.config classes
```

- Add or update tests to freeze:
  - class label order
  - class label values
  - `len(classes)` / `n_outputs` behavior used by the model builders
  - compatibility of `experiments.config.classes` with the new `src/` source of truth

## Hard constraints

- Do not migrate the full `experiments.config` module in this PR.
- Do not change class label order.
- Do not change class label spelling.
- Do not change `n_outputs` for any model.
- Do not change model defaults.
- Do not change model constructor signatures.
- Do not change preprocessing.
- Do not change selected_channels.
- Do not change fold setup or standardization.
- Do not change cropped decoding.
- Do not change W&B behavior.
- Do not change output layout.
- Do not change cache contract.
- Do not change artifact schema.
- Do not migrate data/cache, augmentation, training loop, plotting, or pretraining.
- Do not run full training.
- Do not run pretraining.
- One issue = one PR.

## Required validation

Run focused tests:

```bash
/data/kawamura/envs/bci2020/bin/python -m pytest tests/test_src_experiments_dependency_boundary.py -q
/data/kawamura/envs/bci2020/bin/python -m pytest tests/test_model_registry.py -q
```

Run the new or updated class-label compatibility test.

Also run governance / compatibility guardrails:

```bash
/data/kawamura/envs/bci2020/bin/python -m pytest tests/test_governance_guardrails_docs.py -q
/data/kawamura/envs/bci2020/bin/python -m pytest tests/test_py39_slots_guard.py -q
```

Run dry-run import audit:

```bash
/data/kawamura/envs/bci2020/bin/python scripts/run_experiment.py --config configs/config_smoke_safe.json --dry-run || true
/data/kawamura/envs/bci2020/bin/python -X importtime scripts/run_experiment.py --config configs/config_smoke_safe.json --dry-run 2> importtime.log || true
```

The dry-run may still fail due to the known local cache/config preprocessing mismatch. That is acceptable only if the failure remains cache/config mismatch-related and the import audit completes.

Confirm that the primary dry-run path no longer imports `experiments.models`, and that `src/models/registry.py -> experiments.config classes` has disappeared from the static boundary allowlist.

## Expected result

- `src.models.registry` no longer imports `experiments.config.classes`.
- Class labels have a `src/` source of truth.
- `experiments.config.classes` remains behavior-compatible.
- Static boundary allowlist shrinks by the `src/models/registry.py -> experiments.config classes` entry.
- Model registry keys/defaults remain unchanged.
- No supervised runtime behavior changes.

## Non-goals

- Do not migrate all of `experiments.config`.
- Do not migrate data/cache logic.
- Do not migrate augmentation.
- Do not migrate training loop.
- Do not migrate plotting.
- Do not introduce pretraining changes.
- Do not clean unrelated files.
